internal/backend/connector_updates.go

```diff
@@ -511,3 +499,89 @@ func (user *user) applyMessageDeleted(ctx context.Context, update *imap.MessageD
 
 	return nil
 }
+
+func (user *user) applyMessageUpdated(ctx context.Context, update *imap.MessageUpdated) error {
+	var stateUpdates []state.Update
+
+	if err := user.db.Write(ctx, func(ctx context.Context, tx *ent.Tx) error {
+		var exists bool
+
+		internalMessageID, err := db.GetMessageIDFromRemoteID(ctx, tx.Client(), update.Message.ID)
+		if err != nil && !ent.IsNotFound(err) {
+			return err
+		} else {
+			exists = true
+		}
```

Comment:
`if err != nil && !ent.IsNotFound(err)` --> exists will be marked as true if `err == nil || ent.IsNotFound(err)`, so you will attempt to delete nonexistent messages, right? I guess this is a mistake.

Reply:
Good catch. fixed.